enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value1`\
+   Action = The tag for AWS resource is maintained to `key_infra1 = custom_value`.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action is not applicable.
+
+2) `.spec.platformSpec.aws` take precedence over `.status.platformStatus.aws`. User-defined tags must be merged from `.status.platformStatus.aws` and set to AWS resource.
+   On delete, user-defined must also be removed from `.status.platformStatus.aws`, if present.
+
+### Caveats
+1) User updates the user-defined tag from using external tools when there is an entry in `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is updated from spec, will be reconciled by operators to set value from `.spec.platformSpec.aws.resourceTags`.
+   The user-defined tag will be overwritten with value from `.spec.platformSpec.aws.resourceTags` when there is an update to `.spec.platformSpec.aws.resourceTags`.
+
+   User must handle inconsistencies in `.spec.platformSpec.aws.resourceTags` and user-defined tag value for AWS resource when using multiple tools to manage tags.
+
+   For example,\
+   Edited existing tag using external tool for AWS resource = `key_infra1 = value_comp1`\
+   Previous tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`\
+   Action = Update existing tag with value from `.spec.platformSpec.aws.resourceTags`.\
+   Final tag set to AWS resource = `key_infra1 = value_infra1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) User deletes the user-defined tag from `.spec.platformSpec.aws.resourceTags`
+   The user-defined tag which is removed from spec, will not be reconciled or managed by operators.\
+   User can update user-defined tag key:value using external tool. The user-defined tag will not be overwritten.
+
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value1`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has no user-defined tag with key `key_infra1`\
+   Action = No change in existing user-defined tag.\
+   Final tag set to AWS resource = `key_infra1 = value1`\
+   Event action = No event
+
+3) User sets user-defined tag to delete which is created using external tools.\
+   There is no validation check involved for creator of user-defined tag. Any user-defined tag added by user in `.spec.platformSpec.aws.resourceTags` is considered for create/update/delete accordingly.
+
+4) Any user-defined tag set using `.spec.platformSpec.aws.resourceTags` in `Infrastructure.config.openshift.io/v1` type has scope limited to cluster-level.
```

Comment:
What does it mean to "[have] scope limited to cluster-level"?

Reply:
That means, the tag updated in `.spec.platformSpec.aws.resourceTags` will affect  all AWS resources in the cluster that are being managed by operators.